docs/source-2.0/aws/aws-core.rst

```diff
@@ -1179,6 +1179,290 @@ trait. Setting only the ``requestChecksumRequired`` property of the
+            @required
+            tagKeys: TagKeys
+        }
+        output := { }
+    }
+
+.. _listtags-resource-api:
+
+``ListTagsForResource``
+=======================
+
+The list tags for resource operation retrieves all tag associations on a
+resource.
+
+    * Operation name must case-sensitively match ``ListTagsForResource``.
+    * Must have exactly one input member that targets a string shape and has a
+      member name that matches: ``^([R|r]esource)?([A|a]rn|ARN)$`` to accept
+      the resource ARN.
+    * Must have exactly one output member that targets a list shape, with list
+      member targeting a structure that consists of two members that target a
+      string shape representing the tag key or name and the tag value. This
+      member name must match: ``^[T|t]ag(s|[L|l]ist)$``
+
+The following snippet is an example of the list tags for resource operation and
+its input:
+
+.. code-block:: smithy
+
+    structure Tag { key: String, value: String }
+
+    list TagList { member: Tag }
+
+    operation ListTagsForResource {
+        input := {
+            @required
+            arn: String
+        }
+        output := {
+            @length(max: 128)
+            tags: TagList
+        }
+    }
+
+The following example shows a typical service with the default tagging
+operations. It can be understood that Forecast resource tags are managed
+through the operates attached to the service.
+
+.. code-block:: smithy
+
+    @tagEnabled
+    service Weather {
+        resources: [Forecast]
+        operations: [TagResource, UntagResource, ListTagsForResource]
+    }
+
+    @arn(template: "city/{cityId}/forecast/{forecastId}")
+    @taggable
+    resource Forecast {
+        identifiers: {
+            forecastId: ForecastId
+        }
+    }
+
+
+.. smithy-trait:: aws.api#taggable
+
+--------------------------
+``aws.api#taggable`` trait
+--------------------------
+
+Trait summary
+    Indicates the resource supports AWS tag associations and identifies resource
+    specific operations that perform CRUD on the associated tags. Managing tag
+    associations on a resource through service-wide operations is possible if
+    the resource has an :ref:`aws.api#arn-trait`.
+
+Trait selector
+    ``resource``
+Trait value
+    ``structure``
+
+The ``aws.api#taggable`` trait is a structure that supports the following
+members:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 20 70
+
+    * - Property
+      - Type
+      - Description
+    * - property
+      - ``string``
+      - The name of the resource property representing tags for the resource.
+        Specifying this enables Smithy to understand which resource lifecycle
+        operations operate on tags.
+    * - apiConfig
+      - ``structure``
+      - Configuration structure for resource specific tagging operations, if
+        applicable.
+    * - apiConfig.tagApi
+      - ``ShapeID``
+      - Defines the operation used to create and update tags associations for
```

Comment:
This and the following three properties are required, no?

Reply:
`apiConfig` isn't required. The all three sub-properties are _if_ `apiConfig` is present. I'll adjust the docs to make this clear.

Reply:
Right, the nested properties are required. We usually document these structures separately.